README.md

```diff
@@ -109,6 +109,7 @@ Currently, the following distributions are supported:
 | `oracle` | Oracle JDK | [Link](https://www.oracle.com/java/technologies/downloads/) | [Link](https://java.com/freeuselicense)
 | `dragonwell` | Alibaba Dragonwell JDK | [Link](https://dragonwell-jdk.io/) | [Link](https://www.aliyun.com/product/dragonwell/)
 | `sapmachine` | SAP SapMachine JDK/JRE | [Link](https://sapmachine.io/) | [Link](https://github.com/SAP/SapMachine/blob/sapmachine/LICENSE)
+| `graalvm` | Oracle GraalVM | [Link](https://www.graalvm.org/) | [Link](https://www.oracle.com/downloads/licenses/graal-free-license.html)
```

Comment:
This license information might be incorrect or at least misleading. According to https://www.oracle.com/java/technologies/javase/jdk-faqs.html#GraalVM-licensing (section "What is the licensing for currently available Oracle GraalVM Releases?" and "How long will Oracle use the GFTC and what happens afterwards?") the GraalVM Free Terms and Conditions (GFTC) only apply for one year after the subsequent LTS, so if I understand it correctly
- Oracle GraalVM 17 licensing changed September 2024
- Oracle GraalVM 21 licensing will change September 2026

What exactly does this mean if you use `graalvm` in your workflow?
- Are you still using GraalVM under the GFTC after this time but simply don't receive any updates,
- or are you unwittingly violating the Oracle license?

Related: https://github.com/graalvm/setup-graalvm/issues/107

Reply:
> This license information might be incorrect or at least misleading.

[Script Friendly URLs](https://www.oracle.com/java/technologies/jdk-script-friendly-urls/) are used to pull Oracle GraalVM with `setup-java` and they only provide Oracle GraalVM under the GraalVM Free Terms and Conditions license. So I don't see why the license information would be incorrect.

> What exactly does this mean if you use graalvm in your workflow after these dates?

The action uses the [latest URLs to resolve `java-version: '17'`](https://github.com/actions/setup-java/blob/83a06ff9d9aa70f76a8d73278e646c20b2bf1ae5/src/distributions/graalvm/installer.ts#L85) for Oracle GraalVM and those URLs are no longer available. In practice, this means that `java-version: '17'` no longer works, but you can use `java-version: '17.0.12'` to [pull from the archive](https://github.com/actions/setup-java/blob/83a06ff9d9aa70f76a8d73278e646c20b2bf1ae5/src/distributions/graalvm/installer.ts#L82). By the way, new CPU releases of Oracle GraalVM for JDK 17 are available [here](https://www.oracle.com/java/technologies/downloads/?er=221886#graalvmjava17).

Reply:
> So I don't see why the license information would be incorrect.

I had assumed the license automatically switches from GFTC to the Oracle license, but as you have explained that is not the case. Thanks for the clarification!